### ISSUE TITLE
Always NAN when train with other dataset

### ISSUE THREAD

**Issue by JGIroro** [LiuLei95/PyTorch-Learned-Image-Compression-with-GMM-and-Attention#8]
I want to train a new model with my dataset but always find that the loss becomes larger and larger while PSNR becomes negative. After that, the loss drops to NAN.

**Comment by LiuLei95** [comment]
You can test to use the code in model.py line 102, which can clamp the sigma to (1e^-10, 1e^10) and is commented out by me. The reason for NAN may be the sigma = 0.

**Comment by JGIroro** [comment]
I followed the comment above and used the same dataset and files as yours, finding that although NAN is solved, the loss still becomes larger and larger as well as MSE and bpp.  The only difference between you and me maybe is the environment.  I train on pytorch 1.7 and python 3.6 with cuda 11.0, is it necessary to use the same environment as yours?

**Comment by JGIroro** [comment]
After I use the same environment as yours, the training stage looks good. Thanks a lot!
